### PR TITLE
Maint in progress

### PR DESCRIPTION
**Pull request by MaxJPRey**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -128,45 +129,53 @@ jobs:
...
+      # - name: Upload pytest test results
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     name: pytest-results
+      #     path: junit/test-results.xml
+      #   # Use always() to always run this step to publish test results when there are test failures
+      #   if: ${{ always() }}
+
+      - name: "Check library vulnerabilities with development mode"
+        uses: ansys/actions/check-vulnerabilities@v4.1
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:158]:
```suggestion
        uses: ansys/actions/check-vulnerabilities@v5
```

**Review by Samuelopez-ansys — APPROVED** [review]

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -128,45 +129,53 @@ jobs:
...
-          testenv\Scripts\Activate.ps1
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          python-package-name: 'pyaedt'
+          dev-mode: true
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:163]:
```suggestion
          dev-mode: true
          hide-log: false
```

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -128,45 +129,54 @@ jobs:
...
+          hide-log: false
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:164]:
```suggestion
          hide-log: true
```

File: .github/workflows/unit_tests.yml [hunk 4]
```diff
@@ -128,45 +67,54 @@ jobs:
...
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          python-package-name: 'pyaedt'
+          dev-mode: true
+          hide-log: true
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:102]:
```suggestion
```

File: .github/workflows/unit_tests.yml [hunk 5]
```diff
@@ -128,45 +67,53 @@ jobs:
...
-          testenv\Scripts\Activate.ps1
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          python-package-name: 'pyaedt'
+          dev-mode: true
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:101]:
```suggestion
          dev-mode: false
```

File: .github/workflows/unit_tests.yml [hunk 6]
```diff
@@ -128,45 +67,53 @@ jobs:
...
-          testenv\Scripts\Activate.ps1
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          python-package-name: 'pyaedt'
+          dev-mode: false
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:101]:
```suggestion
          dev-mode: false
          hide-log: true
```

File: .github/workflows/unit_tests.yml [hunk 7]
```diff
@@ -128,45 +70,54 @@ jobs:
...
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
-        run: |
-          testenv\Scripts\Activate.ps1
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:102]:
```suggestion
          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
```

File: .github/workflows/unit_tests.yml [hunk 8]
```diff
@@ -128,45 +70,54 @@ jobs:
...
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
-        run: |
-          testenv\Scripts\Activate.ps1
-          python setup.py sdist
-          python -m pip install twine
-          python -m twine upload --skip-existing dist/*
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+          token: ${{ secrets.PYANSYS_CI_BOT_TOKEN }}
```
Comment (MaxJPRey) [inline .github/workflows/unit_tests.yml:102]:
```suggestion
          token: ${{ secrets.PyAedtVulnerabilities}}
```

**Comment by MaxJPRey** [comment]
The testing phase is completed.

**Pull request closed without merging**